tests/integration/test_wazuh_db/test_wazuh_db.py

```diff
@@ -69,74 +77,165 @@ def restart_wazuh():
+    data = execute_wazuh_db_query(command).split(' ', 1)
+    assert data[0] == 'ok', f"Unable to add agent {agent_id} - {data[1]}"
+
+    command = f"global update-keepalive {update_data}"
+    data = execute_wazuh_db_query(command).split(' ', 1)
+    assert data[0] == 'ok', f"Unable to update agent {agent_id} - {data[1]}"
+
+
+def remove_agent(agent_id):
+    """Function that wraps the needed queries to remove an agent.
+    Args:
+        agent_id(int): Unique identifier of an agent
+    """
+    data = execute_wazuh_db_query(f"global delete-agent {agent_id}").split(' ', 1)
+    assert data[0] == 'ok', f"Unable to remove agent {agent_id} - {data[1]}"
+
+
+# Tests
+
+
+@pytest.fixture(scope='function')
 def pre_insert_agents():
+    """Insert agents. Only used for the global queries"""
     AGENTS_CANT = 14000
     AGENTS_OFFSET = 20
     for id in range(AGENTS_OFFSET, AGENTS_OFFSET + AGENTS_CANT):
-        command = f'global insert-agent {{"id":{id},"name":"TestName{id}","date_add":1599223378}}'
-        receiver_sockets[0].send(command, size=True)
-        response = receiver_sockets[0].receive(size=True).decode()
-        data = response.split(" ", 1)
-        assert data[0] == 'ok', f'Unable to add agent {id}'
+        insert_agent(id)
+
+    yield
+
+    for id in range(AGENTS_OFFSET, AGENTS_OFFSET + AGENTS_CANT):
+        remove_agent(id)
 
-        command = f'global update-keepalive {{"id":{id},"sync_status":"syncreq","connection_status":"active"}}'
-        receiver_sockets[0].send(command, size=True)
-        response = receiver_sockets[0].receive(size=True).decode()
-        data = response.split(" ", 1)
-        assert data[0] == 'ok', f'Unable to update agent {id}'
+
+@pytest.fixture(scope='function')
+def insert_agents_test():
+    """Insert agents. Only used for the agent queries"""
+    agent_list = [1, 2]
+    for agent in agent_list:
+        insert_agent(agent)
+
+    yield
+
+    for agent in agent_list:
+        remove_agent(agent)
 
 
 @pytest.mark.parametrize('test_case',
-                         [case['test_case'] for module_data in module_tests for case in module_data[0]],
+                         [case['test_case'] for module_data in agent_module_tests for case in module_data[0]],
                          ids=[f"{module_name}: {case['name']}"
-                              for module_data, module_name in module_tests
+                              for module_data, module_name in agent_module_tests
                               for case in module_data]
                          )
-def test_wazuh_db_messages(restart_wazuh, clean_registered_agents, configure_sockets_environment, connect_to_sockets_module, test_case):
-    """Check that every input message in wazuh-db socket generates the adequate output to wazuh-db socket
+def test_wazuh_db_messages_agent(restart_wazuh, clean_registered_agents, configure_sockets_environment,
+                                 connect_to_sockets_module, insert_agents_test, test_case):
```

Comment:
After talking with @snaow we have considered that new development or changes into existing tests should follow new repository standards about Wazuh daemons. For now, it is required to use fixtures to start only required daemons, and turn them off when the test is over. That means, that the default status of Wazuh would be off and each test should be responsible for daemons handling.
The reasons for this change are:

- Faster testing
- More encapsulated tests